src/main/java/edu/ucla/library/bucketeer/verticles/ImageWorkerVerticle.java

```diff
@@ -35,22 +53,55 @@ public void start() throws IOException {
             final File tiffFile = new File(json.getString(Constants.FILE_PATH));
             final String imageID = json.getString(Constants.IMAGE_ID);
             final Converter converter = ConverterFactory.getConverter(KakaduConverter.class);
+            final Optional<String> callbackUrlOpt = Optional.ofNullable(json.getString(Constants.CALLBACK_URL));
 
             LOGGER.debug(MessageCodes.BUCKETEER_024, imageID, json.getString(Constants.FILE_PATH));
 
             try {
                 final File jpx = converter.convert(imageID, tiffFile, Conversion.LOSSLESS);
+                final Promise<Void> promise = Promise.promise();
                 final JsonObject message = new JsonObject();
 
                 message.put(Constants.FILE_PATH, jpx.getAbsolutePath());
                 message.put(Constants.IMAGE_ID, jpx.getName());
 
-                sendMessage(message, S3BucketVerticle.class.getName(), Integer.MAX_VALUE);
-            } catch (final InterruptedException | IOException details) {
-                final String message = details.getMessage() != null ? details.getMessage() : LOGGER.getMessage(
-                        MessageCodes.BUCKETEER_030);
+                // Do the conversion before responding back, but respond before doing the S3 upload
+                aMessage.reply(Op.SUCCESS);
+
+                // Handle the S3 storage request's response
+                promise.future().onComplete(upload -> {
+                    if (callbackUrlOpt.isPresent()) {
+                        final WebClient webClient = WebClient.create(vertx);
+                        final boolean callbackUsesSSL;
+                        final String callbackURL;
+
+                        if (upload.succeeded()) {
+                            callbackURL = callbackUrlOpt.get() + "true";
+                        } else {
+                            callbackURL = callbackUrlOpt.get() + "false";
+                        }
+
+                        callbackUsesSSL = callbackURL.startsWith("https") ? true : false;
 
-                LOGGER.error(details, MessageCodes.BUCKETEER_006, message);
+                        LOGGER.debug(MessageCodes.BUCKETEER_515, callbackURL);
+
+                        // Report the success of the S3 upload to whatever initiated the job
+                        webClient.patchAbs(callbackURL).ssl(callbackUsesSSL).send(callback -> {
+                            if (callback.failed()) {
+                                // If the callback isn't listening, just log that as an error
```

Comment:
I think "If no callback URL was provided ..." would be more clear.

Reply:
At this point, we've already checked if the callback URL has been provided (line 73). If we get here, we've attempted to send a result to the callback URL but we weren't able to connect to it (i.e., it wasn't listening as we expected).

Reply:
Oh right. Okay.